### PR TITLE
Feature/ana-1138 : Added examples of generic reconciliation matching rules.    

### PR DESCRIPTION
**Pull request by JTommeyFinbourne**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Added an example for each of the generic reconciliation rules.

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+        {
+            _instrumentLoader = new InstrumentLoader(_apiFactory);
+            _instrumentIds = _instrumentLoader.LoadInstruments();
+        }
+
+        /// <summary>
+        /// Perform a reconciliation on two identical portfolios except for the Address Key for trader name being scope dependent.
+        /// I.e., "Transaction/testPortfolio1/TraderName" = "John Doe" and "Transaction/testPortfolio2/TraderName" = "John Doe"
+        /// Unless a mapping rule is provided between these two address keys in the reconciliation then both keys will appear
+        /// as failures in the aggregation.
```
Comment (ppwfinbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:35]: I'd suggest as "failures to match" to clarify: It's not 'failed' as such, you have done the comparison, just that the result of the comparison is a failure. We use 'failure' very much to mean the request itself failed and that's not the case.

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 2]
```diff
@@ -0,0 +1,773 @@
...
+            var reconciliationResponse = _apiFactory.Api<ReconciliationsApi>().ReconcileGeneric(reconciliation);
+
+            // Get the reconciliation of our equity. 
+            var equityComparison = reconciliationResponse.Comparisons.Single();
+
+            // The trader name properties on the two portfolios have different names and both have been compared upon. 
+            Assert.That(equityComparison.ResultComparison.ContainsKey($"Transaction/{_portfolioOneScope}/TraderName"));
+            Assert.That(equityComparison.ResultComparison.ContainsKey($"Transaction/{_portfolioTwoScope}/TraderName"));
+
+            // Assert that the reconciliation failed on comparing the properties.
```
Comment (ppwfinbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:62]: Again, not quite saying what you mean I think. "Assert that the reconciliation showed that the properties were compared and failed to match".

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 3]
```diff
@@ -0,0 +1,773 @@
...
+            // Check that the property strings where correctly matched.
+            // The difference and comparison dictionaries retain the name of the Right valuation result.
+            Assert.That(equityComparison.Left.ContainsKey($"Transaction/{_portfolioOneScope}/TraderName"));
+            Assert.That(equityComparison.Right.ContainsKey($"Transaction/{_portfolioTwoScope}/TraderName"));
+            Assert.That(equityComparison.Difference.ContainsKey($"Transaction/{_portfolioTwoScope}/TraderName"));
+            Assert.That(!equityComparison.Difference.ContainsKey($"Transaction/{_portfolioOneScope}/TraderName"));
+        }
+
+        /// <summary>
+        /// The default behaviour for comparing numeric values is for exact tolerance. This means only exactly matching
```
Comment (ppwfinbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:91]: A little bit terse. Given this is to explain to a user how the comparison is working, I'd copy or write a bit more of a generic explanation even at the risk of duplication.

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 4]
```diff
@@ -0,0 +1,773 @@
...
+        /// values will match. 
+        /// </summary>
```
Comment (ppwfinbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:93]: Exact match is defined as being numerically equal to machine precision. In the case of decimals that is sufficient. In the case of result with attached units, e.g. currency and amount, the currency would also have to match in addition to exact numeric equality of the amount.

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 5]
```diff
@@ -0,0 +1,773 @@
...
+
+            // Assert that the reconciliation failed for PV, and that the difference is the absolute difference in the PV calculations 
+            Assert.That(equityComparison.Difference["Valuation/PV"], Is.EqualTo(50));
+            Assert.That(equityComparison.ResultComparison["Valuation/PV"].ToString(), Is.EqualTo("Failed"));
+        }
+
+        /// <summary>
+        /// Numeric values can be compared within absolute tolerance. In this example two portfolios provide quotes of £105 and £100
+        /// for an equity. The holding of 10 units results in "Valuation/PV" values of £1050 and £1000 respectively. These can be
+        /// matched within a tolerance of £50.
```
Comment (ppwfinbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:132]: And similarly, £3000 and $3000 dollars would not match.

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 6]
```diff
@@ -0,0 +1,773 @@
...
+                    new PropertyFilter(TestDataUtilities.LusidInstrumentIdentifier, PropertyFilter.OperatorEnum.Equals,
+                        _instrumentIds[0], PropertyFilter.RightOperandTypeEnum.Absolute)
+                },
+                portfolioEntityIds: new List<PortfolioEntityId> {new PortfolioEntityId(scope, code)}
+            );
+
+            return valuationRequest;
+        }
+
+        public static TransactionRequest BuildTransactionRequest(
```
Comment (ppwfinbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:740]: wants a quick comment as is a helper function, also private.

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs (hunk 1 above)
Comment (JTommeyFinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:35]: Have changed

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs (hunk 3 above)
Comment (JTommeyFinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs]: modified to provide more detail

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 7]
```diff
@@ -0,0 +1,841 @@
...
+
+        /// <summary>
+        /// Perform a reconciliation on two identical portfolios except for the Address Key for trader name being scope dependent.
+        /// I.e., "Transaction/testPortfolio1/TraderName" = "John Doe" and "Transaction/testPortfolio2/TraderName" = "John Doe"
+        /// Unless a mapping rule is provided between these two address keys in the reconciliation then both keys will appear
+        /// as fail to match in the aggregation.
+        /// </summary>
+        [LusidFeature("F20-2")]
+        [Test]
+        public void RemappingProperties()
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:39]: rename this to something more descriptive e.g. `Reconcile_Portfolios_By_Property`

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 8]
```diff
@@ -0,0 +1,841 @@
...
+        {
+            var quotePrice = 105m;
+            var units = "GBP";
+            var transactionDate = new DateTimeOffset(2022, 2, 1, 0, 0, 0, TimeSpan.Zero); // date of transaction
+            var traderName = "John Doe";
+
+            // Generate two identical portfolios
+            var valuationRequestOne = GeneratePortfolioTransactions(_portfolioOneScope, _portfolioCode, transactionDate,
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:47]: what is this returning? its not obvious to the reader, add some commentary or split out the steps

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs [hunk 9]
```diff
@@ -0,0 +1,841 @@
...
+
+            // create the reconciliation request
+            var reconciliation = new ReconciliationRequest(valuationRequestOne, valuationRequestTwo);
+            var reconciliationResponse = _apiFactory.Api<ReconciliationsApi>().ReconcileGeneric(reconciliation);
+
+            // Get the reconciliation of the equity. 
+            var equityComparison = reconciliationResponse.Comparisons.Single();
+
+            // The trader name properties on the two portfolios have different addresses and both addresses have been generated in the reconciliation 
+            Assert.That(equityComparison.ResultComparison.ContainsKey($"Transaction/{_portfolioOneScope}/TraderName"));
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:60]: where is this set up?

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs (hunk 8 above)
Comment (JTommeyFinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:47]: will comment to make clearer that this is returning a valuation request for a single equity portfolio

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs (hunk 9 above)
Comment (JTommeyFinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs:60]: All the set up is in a common helper function at the bottom of the file

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs (hunk 7 above)
Comment (JTommeyFinbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/GenericReconciliation.cs]: have renamed the function to Reconcile_By_Remapping_Properties

**Pull request merged**